File: goalbot-backend/src/main/java/com/example/goalbot/entity/Note.java
package com.example.goalbot.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("note")
public class Note {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private String title;

    private String fileName;

    private String summary;

    private String content;

    private String tags;

    private String category;

    @TableField("is_published")
    private Boolean published;

    @TableField("is_official")
    private Boolean official;

    private Integer wordCount;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: goalbot-frontend/src/types/note.ts
export interface Note {
  id: number
  userId: number
  title: string
  fileName?: string | null
  summary?: string | null
  content: string
  tags?: string | null
  category?: string | null
  published: boolean
  official: boolean
  authorName?: string | null
  wordCount: number
  createdAt: string
  updatedAt: string
}

export interface NoteQuery {
  keyword?: string
  category?: string
  published?: boolean
  limit?: number
}

export interface NoteCategory {
  name: string
  count: number
}

export interface NotePayload {
  title: string
  content: string
  tags?: string
  category?: string
  published?: boolean
  official?: boolean
}

export interface PublicNoteItem {
  id: number
  title: string
  category?: string | null
  tags?: string | null
  excerpt: string
  wordCount: number
  updatedAt: string
}

export interface PublicNoteQuery {
  keyword?: string
  category?: string
  descendants?: boolean
  page?: number
  pageSize?: number
}

export interface PublicNotePage {
  items: PublicNoteItem[]
  total: number
  page: number
  pageSize: number
}

export interface PublicNoteNavigation {
  previous: PublicNoteItem | null
  next: PublicNoteItem | null
  position: number
}
